quantestpy/assertion/assert_circuit_equivalent_to_output_qubit_state.py

```diff
@@ -0,0 +1,239 @@
+
+    # get output
+    out_bitstring_actual = \
+        "".join([str(i) for i in pc.qubit_value[output_reg]])
+    out_phase_actual = (pc.qubit_phase[output_reg] / np.pi).tolist()
+    del pc
+
+    if out_bitstring_actual != out_bitstring:
+        return out_bitstring_actual, out_phase_actual
+    elif len(out_phase) > 0 and out_phase_actual != out_phase:
+        return out_bitstring_actual, out_phase_actual
+    else:
+        return None
+
+
+def _draw_circuit(pauli_circuit_org: PauliCircuit,
+                  input_reg: list,
+                  output_reg: list,
+                  in_bitstring: str,
+                  err_msg: str,
+                  val_err_reg: list,
+                  color_phase: bool,
+                  phase_err_reg: list) -> None:
+    # define the circuit
+    pc = copy.deepcopy(pauli_circuit_org)
+    pc.set_qubit_value(input_reg, [int(i) for i in in_bitstring])
+
+    # create an instance of CircuitDrawer
+    gc = PauliCircuitDrawerColorErrorQubit(
+        pc, output_reg, val_err_reg, color_phase, phase_err_reg)
+    gc.set_name_to_reg({"in": input_reg})
+    gc.set_name_to_output_reg({"out": output_reg})
+    gc.draw_circuit()
+    length = len(list(gc.line_id_to_text_whole.values()))
+    fig = gc.create_single_string()
+
+    # show result
+    err_msg_len = len(re.findall(r"\n{1}", err_msg)) + 1
+    print(err_msg)
+    print(fig)
+    input("press enter")
+    for _ in range(err_msg_len+length+1):
+        sys.stdout.write("\033[1A\033[2K")
+    del pc, gc
+    return
+
+
+def assert_circuit_equivalent_to_output_qubit_state(
+        circuit,
+        input_reg: list,
+        output_reg: list,
+        input_to_output: dict,
+        draw_circuit: bool = False):
+    """
+    e.g.
+    input_to_output = {
+        "1000": ("100000", [0.5,0,0,0,0,0]),
+        "1001": "100001"
+    }
+    """
+    quantestpy_circuit = cvt_input_circuit_to_quantestpy_circuit(circuit)
+    pc_org = cvt_quantestpy_circuit_to_pauli_circuit(quantestpy_circuit)
+
+    # check inputs
+    pc_org._assert_is_correct_reg(input_reg)
+    pc_org._assert_is_correct_reg(output_reg)
+    if not isinstance(input_to_output, dict):
+        raise QuantestPyError("input_to_output must be a dict.")
+    if not isinstance(draw_circuit, bool):
+        raise QuantestPyError("draw_circuit must be bool type.")
+
+    len_input_reg = len(input_reg)
+    len_output_reg = len(output_reg)
+
+    for in_bitstring, output in input_to_output.items():
+        # check input type
+        if not isinstance(in_bitstring, str):
+            raise QuantestPyError("Input must be a binary bitstring.")
+        if len(in_bitstring) != len_input_reg:
+            raise QuantestPyError("Input bitstring has an invalid length.")
+
+        # check output type
+        if not isinstance(output, str) and not isinstance(output, tuple):
+            raise QuantestPyError(
+                "Output must be either a binary bitstring or a tuple of"
```

Comment:
Is the following expression correct?
> a tuple of having both a binary bitstring and a list of qubit phases.

I think it is better to use the following two expression:
>a tuple having both a binary bitstring and a list of qubit phases.

Reply:
Done! 👉 https://github.com/QuantestPy/quantestpy/pull/186/commits/1f11dca65e7dae94dc1552126074aab11982fe94